File: push-client-javascript/connector.js
let axios = require('axios')

class Connector {
  constructor(lookupAddress) {
    this.lookupAddress = lookupAddress
    this.host = ''
    this.port = 0
  }

  async lookupConnector() {
      const data = await axios.get(this.lookupAddress)
      const splitResult = data.data.split(':')
      if (splitResult.length !== 2) {
        throw new Error('获取 connector 失败')
      }
      this.host = splitResult[0]
      this.port = parseInt(splitResult[1])
  }

  isAvailable() {
    return this.host && this.port !== 0
  }
}

let conenctor = new Connector('http://localhost:30001')
conenctor.lookupConnector()
console.log(conenctor.isAvailable())

File: push-client-go/client/connector.go
package client

import (
	"fmt"
	"strconv"
	"strings"
)

type Connector struct {
	lookupAddress string
	Host string
	Port int
}
func NewConnector(lookupAddress string) *Connector{
	conn := new(Connector)
	conn.lookupAddress = lookupAddress
	return conn
}
func (conn *Connector) LookupConnector()  {
	resp, err := Get(conn.lookupAddress)
	if err != nil {
		fmt.Println("获取 connector 发生异常", err)
	}
	splitResult := strings.Split(resp, ":")

	if len(splitResult) != 2 {
		fmt.Println("获取 connector 失败", resp)
		return
	}

	conn.Host = splitResult[0]
	conn.Port, _ = strconv.Atoi(splitResult[1])
}
func (conn *Connector) IsAvailable() bool {
	return conn.Host != "" && conn.Port > 0
}
